src/lib/component/autocomplete/autocompleteModel.js

```diff
@@ -0,0 +1,65 @@
+export default class AutocompleteModel {
+  #onTermChange
+  #onItemsChange
+  #onHighlightIndexChange
+  #term = ''
+  #items = []
+  #highlightedIndex = -1
+
+  constructor(onTermChange, onItemsChange, onHighlightIndexChange) {
+    this.#onTermChange = onTermChange
+    this.#onItemsChange = onItemsChange
+    this.#onHighlightIndexChange = onHighlightIndexChange
+  }
+
+  get term() {
+    return this.#term
+  }
+
+  set term(value) {
+    this.#term = value
+
+    if (this.#term.length >= 3) {
+      this.#onTermChange(this.#term)
+    } else {
+      this.#items = [] // Clear items.
+    }
+  }
+
+  get itemsCount() {
+    return this.#items.length
+  }
+
+  set items(value) {
+    this.#items = value
+    this.#highlightedIndex = -1 // Clear highlight.
+    this.#onItemsChange(this.#items)
+  }
+
+  get highlightedIndex() {
+    return this.#highlightedIndex
+  }
+
+  set highlightedIndex(value) {
+    this.#highlightedIndex = value
+    this.#onHighlightIndexChange(this.#highlightedIndex)
+  }
+
+  moveHighlightIndexPrevious() {
+    if (this.highlightedIndex > 0) {
+      this.highlightedIndex--
+    } else if (this.highlightedIndex === 0) {
+      this.highlightedIndex = -1
+    } else {
+      this.highlightedIndex = this.itemsCount - 1
+    }
```

Comment:
```suggestion
    if (this.highlightedIndex >= 0) {
      this.highlightedIndex--
    } else {
      this.highlightedIndex = this.itemsCount - 1
    }
```

と書けるような気がします。
もしそうであれば、ソースコードがNextと対象になって綺麗になります。

Reply:
ありがとうございます。ご提案の形で書けました。
https://github.com/pubannotation/textae/pull/142/commits/5c41a7697fe974f61b4680a14af0429991456368